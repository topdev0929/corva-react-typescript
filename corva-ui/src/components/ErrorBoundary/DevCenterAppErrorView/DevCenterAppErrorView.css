.container {
  display: flex;
  flex-direction: column;
  height: 100%;
  padding: 12px 12px 30px 12px;
}

.content {
  align-items: center;
  color: #9e9e9e;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  text-align: center;
}