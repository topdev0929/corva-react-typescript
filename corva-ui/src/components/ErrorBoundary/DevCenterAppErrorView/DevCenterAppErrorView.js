import PropTypes from 'prop-types';
import EmptyState from '~/components/EmptyState';
import SubtitleWithIntercomLink from '~/components/EmptyView/EmptyAppView/components/ErrorSubtitleWithIntercomLink';

import AppHeader from '../../DevCenter/AppHeader';
import styles from './DevCenterAppErrorView.css';

function DevCenterAppErrorView({ app, openIntercom }) {
  return (
    <div className={styles.container}>
      {app && <AppHeader app={app} />}
      <EmptyState
        title={EmptyState.APP_MESSAGES.internalAppError.title}
        subtitle={<SubtitleWithIntercomLink openIntercom={openIntercom} />}
        image={EmptyState.IMAGES.AppLoadingError}
      />
    </div>
  );
}

DevCenterAppErrorView.propTypes = {
  app: PropTypes.shape({}),
};

DevCenterAppErrorView.defaultProps = {
  app: null,
};

export default DevCenterAppErrorView;
