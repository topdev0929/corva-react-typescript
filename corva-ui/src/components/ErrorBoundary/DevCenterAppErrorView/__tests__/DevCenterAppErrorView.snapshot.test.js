import React from 'react';
import { render } from '@testing-library/react';
import DevCenterAppErrorView from '../DevCenterAppErrorView';

jest.mock('lodash', () => ({
  get: jest
    .fn()
    .mockImplementationOnce(() => 'appProviderUrl')
    .mockImplementationOnce(() => undefined)
    .mockImplementationOnce(() => undefined)
    .mockImplementationOnce(() => undefined),
}));
jest.mock('../../../DevCenter/AppHeader', () => app => (
  <div content={JSON.stringify(app)}>app</div>
));

describe('Snapshot', () => {
  describe('DevCenterAppErrorView', () => {
    beforeEach(jest.restoreAllMocks);
    describe('appProviderUrl', () => {
      it('true', () => {
        const tree = render(<DevCenterAppErrorView app={{}} />).baseElement;
        expect(tree).toMatchSnapshot();
      });

      it('false', () => {
        const tree = render(<DevCenterAppErrorView app={{}} />).baseElement;
        expect(tree).toMatchSnapshot();
      });
    });

    describe('withAppHeader', () => {
      it('true', () => {
        const tree = render(<DevCenterAppErrorView app={{}} withAppHeader={true} />).baseElement;
        expect(tree).toMatchSnapshot();
      });

      it('false', () => {
        const tree = render(<DevCenterAppErrorView app={{}} withAppHeader={false} />).baseElement;
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
