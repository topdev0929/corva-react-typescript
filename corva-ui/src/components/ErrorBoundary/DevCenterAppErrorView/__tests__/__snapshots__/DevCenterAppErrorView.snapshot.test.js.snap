// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot DevCenterAppErrorView appProviderUrl false 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <div
        content="{\\"app\\":{}}"
      >
        app
      </div>
      <div
        class="makeStyles-root-10"
      >
        <div
          class="makeStyles-content-11"
        >
          <div
            class="makeStyles-title-12 makeStyles-title-15"
          >
            App Loading Error
          </div>
          <div
            class="makeStyles-subtitle-13 makeStyles-subtitle-16"
          >
            <span>
              This app failed to load. Try refreshing the page or
               
              <span
                class="makeStyles-subtitle-18"
              >
                contact support
              </span>
              .
            </span>
          </div>
          <img
            alt="error"
            class="makeStyles-image-14 makeStyles-image-17"
            src="/static/media/e934cdf7e3c2d53a46ad705499ddc7ac.svg"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Snapshot DevCenterAppErrorView appProviderUrl true 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <div
        content="{\\"app\\":{}}"
      >
        app
      </div>
      <div
        class="makeStyles-root-1"
      >
        <div
          class="makeStyles-content-2"
        >
          <div
            class="makeStyles-title-3 makeStyles-title-6"
          >
            App Loading Error
          </div>
          <div
            class="makeStyles-subtitle-4 makeStyles-subtitle-7"
          >
            <span>
              This app failed to load. Try refreshing the page or
               
              <span
                class="makeStyles-subtitle-9"
              >
                contact support
              </span>
              .
            </span>
          </div>
          <img
            alt="error"
            class="makeStyles-image-5 makeStyles-image-8"
            src="/static/media/e934cdf7e3c2d53a46ad705499ddc7ac.svg"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Snapshot DevCenterAppErrorView withAppHeader false 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <div
        content="{\\"app\\":{}}"
      >
        app
      </div>
      <div
        class="makeStyles-root-28"
      >
        <div
          class="makeStyles-content-29"
        >
          <div
            class="makeStyles-title-30 makeStyles-title-33"
          >
            App Loading Error
          </div>
          <div
            class="makeStyles-subtitle-31 makeStyles-subtitle-34"
          >
            <span>
              This app failed to load. Try refreshing the page or
               
              <span
                class="makeStyles-subtitle-36"
              >
                contact support
              </span>
              .
            </span>
          </div>
          <img
            alt="error"
            class="makeStyles-image-32 makeStyles-image-35"
            src="/static/media/e934cdf7e3c2d53a46ad705499ddc7ac.svg"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Snapshot DevCenterAppErrorView withAppHeader true 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <div
        content="{\\"app\\":{}}"
      >
        app
      </div>
      <div
        class="makeStyles-root-19"
      >
        <div
          class="makeStyles-content-20"
        >
          <div
            class="makeStyles-title-21 makeStyles-title-24"
          >
            App Loading Error
          </div>
          <div
            class="makeStyles-subtitle-22 makeStyles-subtitle-25"
          >
            <span>
              This app failed to load. Try refreshing the page or
               
              <span
                class="makeStyles-subtitle-27"
              >
                contact support
              </span>
              .
            </span>
          </div>
          <img
            alt="error"
            class="makeStyles-image-23 makeStyles-image-26"
            src="/static/media/e934cdf7e3c2d53a46ad705499ddc7ac.svg"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
